addons/account/models/account_move.py

```diff
@@ -2270,6 +2270,13 @@ def action_post(self):
     def js_assign_outstanding_line(self, line_id):
         self.ensure_one()
         lines = self.env['account.move.line'].browse(line_id)
+        if not lines.currency_id and self.currency_id != self.company_id.currency_id and not lines.matched_debit_ids and not lines.matched_credit_ids:
```

Comment:
` not lines.matched_debit_ids and not lines.matched_credit_ids` is incorrect and must be removed. If the line is already reconciled, it means there is a bug somewhere allowing the user to do something leading to a wrong behavior.

Reply:
Removed.